### ISSUE TITLE
Low Priority: Non-consecutive dates as input to index date range 

### ISSUE THREAD

**Issue by annaschultze** [opensafely-core/cohort-extractor#450]
Low priority enhancement, but it's often useful to extract exactly the same cohort at different - non consecutive - dates, for example 1 Feb 2019, 1 Feb 2020 and 1 Sep 2021 to analyse characteristics before the pandemic, before the 1st wave and before the 2nd wave. 

There are work-arounds to implement this more elegantly than just copying and pasting a study definition, but a simple solution would also be to allow `index-date-range` in the cohortextractor to take a list of nonconsecutive values as input. This might make the intention of extracting multiple cohorts more transparent and would likely be easier for non-Python versed people than the `common_variables.py` solution proposed by @alexwalkercebm.

**Comment by alexwalkercebm** [comment]
It's probably possible to do this in a slightly hacky way already.

If you have two separate generate cohort actions, and use `--index-date-range "2020-03-01 to 2020-03-01 by month"` and `--index-date-range "2020-09-01 to 2020-09-01 by month"`
I haven't tested this, but the first action should then make an output called `input-2020-03-01.csv` and the second `input-2020-09-01.csv`.

**Comment by evansd** [comment]
`--index-date-range` can actually take a single date already so just `--index-date-range "2020-03-01"` ought to do the right thing.

I don't think it would be massively difficult to make it take a comma separated list of dates are something like that.

That said, doing it using separate actions as you suggested would probably be faster because the extracts would run in parallel.

**Comment by alexwalkercebm** [comment]
Oh great, it's not even a hacky solution then. I don't think a comma separated list would add much.

There's already (maybe distant) plans to loop over the same action with different arguments anyway right?

**Comment by annaschultze** [comment]
Amazing, thanks - this will help simplify some of my existing code quite a bit. I'll close the issue.
